sdk/cosmos/azure-cosmos-spark_3-1_2-12/src/main/scala/com/azure/cosmos/spark/PartitionMetadataCache.scala

```diff
@@ -182,9 +182,25 @@ private object PartitionMetadataCache extends CosmosLoggingTrait {
           assertNotNull(lastDocumentCount.get, "lastDocumentCount"),
           assertNotNull(lastTotalDocumentSize.get, "lastTotalDocumentSize"),
           assertNotNullOrEmpty(lastContinuationToken.get, "continuationToken")
-        )
+        ))
+      })
+      .onErrorResume((throwable: Throwable) => {
+        throwable match {
+          case cosmosException: CosmosException =>
+            if (tolerateNotFound &&
+              cosmosException.getStatusCode == HttpConstants.StatusCodes.NOTFOUND &&
```

Comment:
you can factor out the code to `Exceptions` helper class, we have similar other methods.

https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/cosmos/azure-cosmos-spark_3-1_2-12/src/main/scala/com/azure/cosmos/spark/Exceptions.scala

Reply:
ACK - Done